static/app/components/nav/overlay.tsx

```diff
@@ -0,0 +1,60 @@
+import {type ComponentProps, forwardRef} from 'react';
```

Comment:
minor nit: You pretty much never need to import react types, since they're all available on the `React` global. So this can just be `React.ComponentProps`

Reply:
Hmm - I would actually say it's better to deconstruct it from the import.

By doing that it creates a clear dependency for those properties. I am also extremely weary of the global space, it can often be polluted and cause more harm than good to rely on those.

Reply:
I always forget about the `React` global! Updated in 6a06fcea1e5

Reply:
> I am also extremely weary of the global space, it can often be polluted and cause more harm than good to rely on those.

I certainly agree that global space isn't usually great. But I think for the case of react types, especially because they're namespaced under `React`. This is convenient and has very few downsides.